### PR TITLE
[TEST] Add an end-to-end test

### PR DESCRIPTION
**Pull request by rohitpaulk**
Based on https://github.com/segmentio/analytics-node/pull/134.

The `RUNSCOPE_TOKEN` env var has to be added for this to work, tests will fail until that is done. Here's an example of passing tests on my fork: https://github.com/rohitpaulk/analytics-ruby/pull/10

### CODE REVIEW

File: spec/segment/analytics/e2e_spec.rb [hunk 1]
```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+module Segment
+  # End-to-end tests that send events to a segment source and verifies that a
+  # webhook connected to the source (configured manually via the app) is able
+  # to receive the data sent by this library.
+  describe 'End-to-end tests', e2e: true do
+    WRITE_KEY = 'qhdMksLsQTi9MES3CHyzsWRRt4ub5VM6'
+    RUNSCOPE_BUCKET_KEY = 'pwb8mcmfks0f'
```
Comment (f2prateek) [inline spec/segment/analytics/e2e_spec.rb]: Can you add a comment mentioning this bucket can be viewed at `https://www.runscope.com/stream/umkvkgv7ndby`

Comment (f2prateek) [reply spec/segment/analytics/e2e_spec.rb]: (`umkvkgv7ndby` and not `pwb8mcmfks0f` - sorry about the mixup!)

File: spec/segment/analytics/e2e_spec.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
+require 'spec_helper'
+
+module Segment
+  # End-to-end tests that send events to a segment source and verifies that a
+  # webhook connected to the source (configured manually via the app) is able
+  # to receive the data sent by this library.
+  describe 'End-to-end tests', e2e: true do
+    WRITE_KEY = 'qhdMksLsQTi9MES3CHyzsWRRt4ub5VM6'
```
Comment (f2prateek) [inline spec/segment/analytics/e2e_spec.rb]: Could you add a comment that this writeKey corresponds to https://app.segment.com/segment-libraries/sources/analytics_ruby_e2e_test/overview

File: Rakefile [hunk 3]
```diff
@@ -4,6 +4,7 @@ default_tasks = []
 
 RSpec::Core::RakeTask.new(:spec) do |spec|
   spec.pattern = 'spec/**/*_spec.rb'
+  spec.rspec_opts = "--tag ~e2e" if ENV["RUN_E2E_TESTS"] != "true"
```
Comment (f2prateek) [inline Rakefile]: We should move this to Travis (like the runscope token). I believe non-owners won't be able to get the Runscope token anyway.

Comment (f2prateek) [reply Rakefile]: I've added RUN_E2E_TESTS as a variable in travis https://cloudup.com/cmDLAS18POA

Comment (rohitpaulk) [reply Rakefile]: I placed this check here to ensure that `bundle exec rake` works on a local system without hitting the network - less friction for an external contributor trying to run tests on their machine. 

> I believe non-owners won't be able to get the Runscope token anyway.

Ah, you're right - I've changed this to only run the end-to-end tests if travis secure env vars are present (master builds, or PRs from owners). https://github.com/segmentio/analytics-ruby/pull/137/commits/6b2983ce92c4ed72a8a037cfb12f26e326d67c78

File: .travis.yml [hunk 4]
```diff
@@ -14,3 +11,10 @@ rvm:
 
 before_install:
   - gem install bundler
+
+script:
+  - if [ "$TRAVIS_SECURE_ENV_VARS" != "false" ]; then
+      RUN_E2E_TESTS=true bundle exec rake
```
Comment (rohitpaulk) [inline .travis.yml]: @f2prateek For this to work, we'll have to remove the `RUN_E2E_TESTS` variable added to travis. 

Either that, or we set both `RUN_E2E_TESTS` **and** `RUNSCOPE_TOKEN` as encypted variables and remove this check.

File: .travis.yml [hunk 5]
```diff
@@ -14,3 +11,10 @@ rvm:
 
 before_install:
   - gem install bundler
+
+script:
+  - if [ "$TRAVIS_SECURE_ENV_VARS" != "false" ]; then
```
Comment (rohitpaulk) [inline .travis.yml]: @f2prateek For this to work, we'll have to remove the `RUN_E2E_TESTS` variable added to travis. 

Either that, or we set both `RUN_E2E_TESTS` **and** `RUNSCOPE_TOKEN` as encypted variables and remove this check.

Comment (f2prateek) [reply .travis.yml]: I can update the  RUN_E2E_TESTS to be encrypted. That seems easiest - also makes it easy to run the tests locally when needed.

Comment (rohitpaulk) [reply .travis.yml]: Cool, I'll remove the check then

Comment (f2prateek) [reply .travis.yml]: Added `RUN_E2E_TESTS` as an encrypted variable now.

**Comment by rohitpaulk** [comment]
Code updated. The E2E test should **not** be triggered in this PR, as it is from my fork. I've opened https://github.com/segmentio/analytics-ruby/pull/139 to actually run the E2E test.

Will close this as soon as the tests pass.

**Pull request closed without merging**